libeos-payg/manager.c

```diff
@@ -617,23 +644,32 @@ set_expiry_time (EpgManager *self,
   /* Clamp to the end of time instead of overflowing. */
   self->expiry_time_secs = (base_secs <= G_MAXUINT64 - span_secs) ? base_secs + span_secs : G_MAXUINT64;
 
-  /* Set the expiry timer. g_timeout_source_new_seconds() takes a #guint, and
+  /* Set the expiry timer. epg_clock_source_new_seconds() takes a #guint, and
    * @span is a #guint64. However, the maximum span is 365 days, which is
-   * representable in 32 bits. We don’t set a timer for infinite periods.
-   *
-   * FIXME: For the moment, poll every 60s until the expiry time is reached, so
-   * we don’t have to worry about recalculating the timeout period after
-   * resuming from suspend, or if the system clock or timezone changes.
-   * See: https://phabricator.endlessm.com/T22074 */
+   * representable in 32 bits. We don’t set a timer for infinite periods. */
```

Comment:
I know this is existing code but it's not actually true that the maximum time to expiry is 365 days. The maximum period for a code is indeed 365 days, or 2 ** 24.910496345705575 seconds, but there are 2 ** 8 possible codes for each period, and @treitter is working away at making it easy to add a code before the old one has expired. If you add 137 365-day codes, the expiry time is more than 2 ** 32 seconds into the future, so the assertion below will fail.

Reply:
Added a fixup commit to clamp to G_MAXUINT